### PR TITLE
94 x htt

### PR DESCRIPTION
**Pull request by GaelTouquet**

### CODE REVIEW

**Comment by cbernet** [comment]
@steggema would you like to review?

**Comment by steggema** [comment]
so the reason you remove the muon scale factors is that they only apply to the single muon trigger + related offline selection and ID? 
it could still be good to keep them (and maybe just save their values in the output trees, not multiply them with the general event weight) if one wants to start doing data-MC comparisons soon. if the purpose is only synchronisation it's ok to drop them for now

**Comment by GaelTouquet** [comment]
yes the purpose is to drop them just for now since they are not final and the bins out of range messages slow down heppy

**Comment by steggema** [comment]
With "not final" you mean that these are not going to be the final scale factors we use? (since they are the final scale factors the muon POG will provide)
If it's only the error messages that slow down heppy, we should reduce/omit those

File: H2TauTau/python/proto/weights/ScaleFactor.py [hunk 1]
```diff
@@ -185,11 +185,11 @@ def getFactorJson(self, pt, eta):
                     pt_low, pt_high = re.findall("\d+\.\d+", pt_key)
                     if pt >= float(pt_low) and pt < float(pt_high):
                         return sub_eff[pt_key][u'value']
-                else:
-                    print 'pT out of range, using last pT bin'
-                    return sub_eff[pt_key][u'value']
-        else:
-            print 'JSON-based scale factor: eta out of range, last eta_high = ', eta_high, 'eta = ', eta
+                    else:
```
Comment (steggema) [inline H2TauTau/python/proto/weights/ScaleFactor.py]: I actually don't understand why you change the indentation here. This should only be reached if the for-loop completes without finding a pT match, so the implementation looks OK to me.

File: H2TauTau/python/proto/weights/ScaleFactor.py [hunk 2]
```diff
@@ -185,11 +185,11 @@ def getFactorJson(self, pt, eta):
...
+                        print 'pT out of range, using last pT bin'
+                        return sub_eff[pt_key][u'value']
+            else:
```
Comment (steggema) [inline H2TauTau/python/proto/weights/ScaleFactor.py]: Same here

File: H2TauTau/python/proto/samples/fall17/triggers_tauMu.py [hunk 3]
```diff
@@ -37,18 +40,20 @@
...
     # 'HLT_IsoMu21_eta2p1_LooseIsoPFTau20_SingleL1_v5',
+    'HLT_IsoMu20_eta2p1_LooseChargedIsoPFTau27_eta2p1_CrossL1_v*',
     # 'HLT_VLooseIsoPFTau140_Trk50_eta2p1_v5',
     # 'HLT_VLooseIsoPFTau120_Trk50_eta2p1_v5',
     ]
 
 mc_triggerfilters = [
     TFM(leg1_names=['hltL3crIsoL1sSingleMu22L1f0L2f10QL3f24QL3trkIsoFiltered0p07'], leg2_names=[], triggers=['HLT_IsoMu24_v*']), # IsoMu24
-    TFM(leg1_names=['hltL3crIsoL1sMu22Or25L1f0L2f10QL3f27QL3trkIsoFiltered0p07'], leg2_names=[], triggers=['HLT_IsoMu27_v*']), # IsoMu24
+    TFM(leg1_names=['hltSelectedPFTau27LooseChargedIsolationAgainstMuonL1HLTMatched','hltOverlapFilterIsoMu20LooseChargedIsoPFTau27L1Seeded'], leg2_names=['hltL3crIsoL1sMu18erTau24erIorMu20erTau24erL1f0L2f10QL3f20QL3trkIsoFiltered0p07','hltOverlapFilterIsoMu20LooseChargedIsoPFTau27L1Seeded'], triggers=['HLT_IsoMu20_eta2p1_LooseChargedIsoPFTau27_eta2p1_CrossL1_v*'], match_all_names_leg1=True, match_all_names_leg2=True), # IsoMu19_eta2p1_LooseIsoPFTau20
```
Comment (cbernet) [inline H2TauTau/python/proto/samples/fall17/triggers_tauMu.py:55]: please cut long lines for readability in the future.
could have cut between the arguments of TFM, and between the elements of e.g. leg1_names. 
I will merge anyway

**Pull request merged**